File: spring-utils/src/main/java/chatox/platform/log/LogExecution.java
package chatox.platform.log;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.METHOD, ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface LogExecution {
    boolean logReturnValue() default true;
    LogLevel returnValueLogLevel() default LogLevel.DEBUG;
    boolean logParameters() default true;
    LogLevel parametersLogLevel() default LogLevel.DEBUG;
    LogLevel executionLogLevel() default LogLevel.DEBUG;
    String displayedMethodName() default "";
    String[] excludedMethods() default {};
    boolean logOnlyPublicMethods() default true;
    boolean logError() default true;
    LogLevel errorLogLevel() default LogLevel.ERROR;
}

File: chat/src/main/kotlin/chatox/chat/support/log/LogExecution.kt
package chatox.chat.support.log

@Target(allowedTargets = [AnnotationTarget.FUNCTION, AnnotationTarget.CLASS])
@Retention(AnnotationRetention.RUNTIME)
annotation class LogExecution(
        val logReturnValue: Boolean = true,
        val logReturnValueLevel: LogLevel = LogLevel.DEBUG,
        val logParameters: Boolean = true,
        val logParametersLogLevel: LogLevel = LogLevel.DEBUG,
        val logExecutionLogLevel: LogLevel = LogLevel.DEBUG,
        val logError: Boolean = true,
        val logErrorLogLevel: LogLevel = LogLevel.ERROR,
        val methodName: String = "",
        val excludedMethods: Array<String> = [],
        val logOnlyPublicMethods: Boolean = true
)
